File: submissions/1894-merge-strings-alternately/solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        char[] chars1 = word1.toCharArray();  
        char[] chars2 = word2.toCharArray();

        int maxLength = Math.max(chars1.length, chars2.length);
        int minLength = Math.min(chars1.length, chars2.length);

        char[] largerChars = chars1.length > chars2.length ? chars1 : chars2;

        StringBuilder resultBuilder = new StringBuilder();

        for (int i = 0; i < minLength; i++) {
            resultBuilder.append(chars1[i]);
            resultBuilder.append(chars2[i]);
        }
        
        for (int i = minLength; i < maxLength; i++) {
            resultBuilder.append(largerChars[i]);
        }

        return resultBuilder.toString();

    }
}

File: submissions/1894-merge-strings-alternately/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        w1Length = len(word1)
        w2Length = len(word2)
        result = ""

        for i in range(min(w1Length, w2Length)):
            result += word1[i]
            result += word2[i]

        if w1Length > w2Length:
            result += word1[w2Length:]
        elif w2Length > w1Length:
            result += word2[w1Length:]

        return result
        
